src/serac/physics/nonlinear_solid.hpp

```diff
@@ -188,6 +188,11 @@ class NonlinearSolid : public BasePhysics {
    */
   void advanceTimestep(double& dt) override;
 
+  /**
+   * @brief Get an mfem::Operator that calculates the residual at the current outputState
+   */
+  mfem::Vector getOutputResidual();
```

Comment:
What do you think about adding the two methods to the `BasePhysics` class? I think it's valuable to keep consistent naming in the physics modules. I'm not sure though if they should be required in an implementation.

Reply:
I added it to BasePhysics and then implemented stubs/errors the "mfem-way" for non-implemented physics.